open-craft/learning-paths-plugin
# [BB-9214] Implement pathway progress api

## Description
This PR adds an endpoint that returns the aggregate progress of the requested user in a learning path. It in turn depends on the completion aggregator API for fetching the completion percentages of the individual courses that make up a learning path.

## Supporting Information
OpenCraft Internal Jira ticket: [BB-9214](https://tasks.opencraft.com/browse/BB-9214)

## Testing Instructions
1. Set up devstack.
2. Clone this PR branch repository to `<devstack root>/src`
3. Enter lms shell: `$ make dev.shell.lms`
4. Install this plugin: `# pip install -e /edx/src/learning-paths-plugin/`
5. Run migrations: `# ./manage.py lms migrate learning_paths`
6. Run completion aggregator migrations: `# ./manage.py lms migrate completion_aggregator`
7. Once the plugin is installed, create a learning path http://localhost:18000/admin/learning_paths/learningpath/ with some test courses and enroll a user into it.
8. Open the learning path and note the id from the address bar. It will be in the form of `admin/learning_paths/learningpath/<id>/change/`.
9. Login to lms as the enrolled user.
10. While being logged in, test the learning path progress using http://localhost:18000/api/v1/learning-paths/1/progress/.
11. Go back to lms and progress through the course(s).
12. Verify that the progress of the learning path is also updated.
